git-branchless-lib/src/git/repo.rs

```diff
@@ -565,26 +565,41 @@ impl Repo {
         Ok(Config::from(config))
     }
 
+    /// Get the directory where all repo-specific git-branchless state is stored.
+    pub fn get_branchless_dir(&self) -> Result<PathBuf> {
+        let maybe_worktree_parent_repo = self.open_worktree_parent_repo()?;
```

Comment:
A couple of weeks ago I had a hunch that the issue I described in #928 was being caused by `git-branchless` using separate directories for each worktree (sorry for not being more responsive in that discussion, it's been a busy few months!).

I even pulled down the code and started digging through things to see if I could figure out where the changes might need to be made. One thing I did come across was `git_repository_commondir` from `libgit2`, which seems like it may be a built-in version of `open_worktree_parent_repo`, but I'm not 100% sure. Unfortunately, the `git2` crate doesn't seem to expose it and that's as far as I got before I ran out of time that day 😝

https://libgit2.org/libgit2/#HEAD/group/repository/git_repository_commondir

Reply:
Ah, commondir does sound like what I'm looking for (also available on the CLI as `git rev-parse --git-common-dir`).